redhat-appstudio/service-provider-integration-operator
# configurable Vault path prefix

### What does this PR do?
Option to configure vault path prefix. Default is still `spi` so without explicitly setting it, everything should stay same.
 - new cli arg for both operator and oauth `--vault-data-path-prefix` env `VAULTDATAPATHPREFIX`
 - new configurable `hack/vault-init.sh` env variable `SPI_DATA_PATH_PREFIX`

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-259

### How to test this PR?


images:
 - `quay.io/mvala/service-provider-integration-operator:svpi259vaultPrefix`
 - `quay.io/mvala/service-provider-integration-oauth:svpi259vaultPrefix`

Let's say you want to have data at `some/test/path`

1. deploy with images ^ with env variable `SPI_DATA_PATH_PREFIX` set to some path (for example on minikube something like `make deploy_minikube SPI_IMG_BASE=quay.io/mvala TAG_NAME=svpi259vaultPrefix SPI_DATA_PATH_PREFIX=some/test/path`)
2. set env variable in shared configmap for SPI pods `kubectl patch cm -n spi-system spi-shared-environment-config --type json --patch '[{"op": "add", "path": "/data/VAULTDATAPATHPREFIX", "value": "some/test/path"}]'`
3. try some action that writes data to Vault, like manual token upload or anything
4. to check you're really storing data at configured path, you can either connect to vault pod shell and use `vault` cli, or enable debug logs in SPI pods and see vault path in logs
